docs/search_syntax.md

```diff
@@ -0,0 +1,228 @@
+```python
+run.duration <= 111111111
+```
+
+#### Example with ```run.archived``` (boolean)
+Select only the runs where the archived attribute is true
+```python
+run.archived == True
+```
+
+Select only the runs where the archived attribute is not true
+```python
+run.archived != True
+```
+
+### Run parameters
+Run parameters could be accessed via attributes.
+```python
+run.hparams.param1 >= 64
+```
+![FastTrackML Run List, param filter](images/runs_param_filter.png)
+
+### Filtering Runs with Unset Attributes
+
+To filter runs based on whether an attribute is not set, you can use the following syntax:
+
+```python
+run.attribute is None
+```
+This expression will return runs for which the specified attribute is not defined.
+
+Showing all the runs
+![FastTrackML Run List](images/runs.png)
+
+Showing only the runs where param1 is not set
+![FastTrackML Run List of not set param](images/runs_none_param.png)
+
+### Filter Runs using Regular Expressions
+- ``` .match() ```
+- ``` .search() ```
+
+Match finds an exact match at the beginning of a string.
+![FastTrackML Run filter using regular expression match](images/runs_regular_expression_match.png)
+
+Search looks for a pattern anywhere in the string.
+![FastTrackML Run filter using regular expression match](images/runs_regular_expression_search.png)
+
+#### Example with ```metric.name``` (string)
+Select only the metrics where the name exactly matches "TestMetric1"
+```python
+metric.name == "TestRun1"
+```
+
+Select only the metrics where the name is different from "TestMetric1"
+```python
+metric.name != "TestRun1"
+```
+
+Select only the metrics where "Run1" is contained in the run name
+```python
+"Metric1" in metric.name
+```
+
+Select only the metrics where the name starts with "Test"
+```python
+metric.name.startswith('Test')
+```
+
+Select only the metrics where the name ends with "Metric1"
+```python
+metric.name.endswith('Metric1')
+```
+
+#### Example with ```metric.last``` (numeric)
+
+Select only the metrics where the last value is exactly 1.1
+
+```python
+metric.last == 1.1
+```
+
+Select only the metrics where the last value is not 1.1
+```python
+metric.last != 1.1
+```
+
+Select only the metrics where the last value is greater than 1.1
+```python
+metric.last > 1.1
+```
+
+Select only the metrics where the duration is greater or equal to 1.1
+```python
+metric.last >= 1.1
+```
+
+Select only the metrics where the last value is less than 1.1
+```python
+metric.last < 1.1
+```
+
+Select only the metrics where the last value is less or equal to 1.1
+```python
+metric.last <= 1.1
+```
+
+### Filter Metrics by run
+You can also filter the metrics by combining  metric attributes with run attributes.
+
+Showing the metrics with the last value greater than 6 belonging to a run with the name that starts with marvelous.
+
+![FastTrackML Metric List filter by metric and run attributes](images/metrics_filter_by_run.png)
```

Comment:
what do you think to a like a complex query with different operators and cases?

Reply:
Yep